sources/web/datalab/settings.ts

```diff
@@ -396,6 +399,21 @@ export function createHandler(): http.RequestHandler {
   return requestHandler;
 }
 
+function renameBadUserSettings(settingsPath: string) {
+  let newPath = settingsPath + ".bad";
+  let n = 0;
+  while (fs.existsSync(newPath)) {
```

Comment:
Do we need a limit on this? Since we still don't know what's causing the bug, if this happens frequently enough we'll fill up the file system with these files.
Maybe we can just use one -bad name and override it?